### PR TITLE
[WIP]164 project search functionality fixed

### LINKED ISSUE #164: we lose our search if we click on a project and navigate back
## Expected Behavior

A user's search should persist navigating to project's info page/until they clear their search
## Current Behavior

At the moment, we can search for projects based on the programming language they use. 
If one clicks on a project to see a more detailed view of its info, then navigates back to `/projects`, they lose their search term
## Possible Solution

save the searched-for language to Redux
## Steps to Reproduce (for bugs)

1. visit `/projects`
2. search for a language in the dropdown, e.g. `JavaScript`
3. click on one of the project's, which will take you to `/projects/<project-slug>`
4. click on the back button in your browser

## Context

Users expect that when they search for a term, that search will be preserved until they clear it.

### PR DESCRIPTION
**Pull request by nataliyah123**
fixes #164

I had a detached head and a  stuck rebase, which was causing the problems in the older branch .i have fixed that however few files did escape the rebase which i manually added. however i cannot tell which branch will work fine for others . so plz let me know and will remove the other branch

. I have made the requested changes from older in this branch

### CODE REVIEW

File: package.json~ [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+  "cypress-cucumber-preprocessor": {
+    "nonGlobalStepDefinitions": true
+  },
+  "dependencies": {
+    "axios": "^0.18.0",
+    "babel-plugin-transform-object-rest-spread": "^6.26.0",
+    "dotenv-webpack": "^1.7.0",
+    "global": "^4.3.2",
+    "izitoast": "^1.4.0",
+<<<<<<< HEAD
```
Comment (mattwr18) [inline package.json~]: @nataliyah123... there are several places where you didn't work through the merge conflicts?
a good way to catch these is to run `yarn test` locally cause that will run `standard` and fail if you've forgotten any.

File: src/actions/getEventInfoAction.js~ [hunk 2]
```diff
@@ -0,0 +1,49 @@
+import axios from 'axios'
+import { GET_PROJECT_INFO } from '../types'
+
+<<<<<<< HEAD
```
Comment (mattwr18) [inline src/actions/getEventInfoAction.js~]: same as above

File: src/components/App.js~ [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+import PremiumMembershipPage from '../containers/PremiumMembershipPage'
+import PremiumMobMembershipPage from '../containers/PremiumMobMembershipPage'
+import PremiumF2FMembershipPage from '../containers/PremiumF2FMembershipPage'
+import GettingStartedPage from '../containers/GettingStartedPage'
+import EventsList from '../containers/EventsList'
+import EventInfo from '../containers/EventInfo'
+<<<<<<< HEAD
+import CreateEventPage from '../containers/CreateEventPage'
+=======
+>>>>>>> Add EventInfo and EventSummary Components
```
Comment (mattwr18) [inline src/components/App.js~]: same as above

**Review by mattwr18 — CHANGES_REQUESTED** [review]

File: package.json~ (hunk 1 above)
Comment (nataliyah123) [reply package.json~]: i had run the yarn test before i pushed it and i got all cleared test . but when i pushed it though i had done rebasing but it showed merged conflict in two files App.js and initialstate so i fixed them here . but it is still showing error. i will try to fix them as well now

**Pull request closed without merging**